app/domain/authentication/authn_gcp/authentication_request.rb

```diff
@@ -0,0 +1,30 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible for retrieving the correct value from the GCP token
+    # of the requested attribute.
+    class AuthenticationRequest
+      def initialize(decoded_token:)
+        @decoded_token = decoded_token
+      end
+
+      def retrieve_attribute(attribute_name)
```

Comment:
Think about it as an interface.
How would you define this method's signature?
As I see it, every request has attributes.
For GCP, these attributes are retrieved from the token claims, but that's internal implementation.
The caller doesn't care how and from where the values are brought.

Reply:
@abrahamko do we have some way to enforce it? so that an `AuthenticationRequest` class will always implement this method?

also - what happens if it doesn't? which error will be raised?

Reply:
1st question: AFAIK no. Ruby does not have such mechanisms.
2nd question: If it doesn't, (or if the args does not match) it throws NoMethodError.

To handle both cases, integration tests will cover it.